lib/amqp/channel.ex

```diff
@@ -45,8 +46,11 @@ defmodule AMQP.Channel do
       {:ok, chan_pid} ->
         {:ok, %Channel{conn: conn, pid: chan_pid, custom_consumer: custom_consumer}}
 
-      error ->
-        error
+      {:error, error} ->
+        {:error, error}
+
+      :closing ->
+        {:error, :closing}
```

Comment:
Not 100% sure what the policy in this repo is, but this way, we ensure we match only on tuple or closing atom, ensuring a tuple is returned.

If necessary, I can reintroduce the "catch all" error clause.

Reply:
Is it possible to produce a test for this reliably, without mocks?

Reply:
```elixir
error -> {:error, error}
```

should cover the case. like `close/1`.

I was thinking to pass a closed connection but not sure if it gives an error expected. if not, no worries about the test i can work out.

Reply:
Ah, sorry. I'm really bad at reading the underlying erlang code. I see what you mean now. I pushed another commit